servicetalk-examples/http/debugging/src/main/java/io/servicetalk/examples/http/debugging/DebuggingExampleUrlClient.java

```diff
@@ -0,0 +1,255 @@
+ * 2023-08-29 18:02:12,536 servicetalk-global-io-executor-1-2 [TRACE] servicetalk-examples-wire-logger - [id: 0xc0b781e2, L:/127.0.0.1:63690 ! R:localhost/127.0.0.1:8080] INACTIVE
+ * 2023-08-29 18:02:12,537 servicetalk-global-io-executor-1-2 [TRACE] servicetalk-examples-wire-logger - [id: 0xc0b781e2, L:/127.0.0.1:63690 ! R:localhost/127.0.0.1:8080] UNREGISTERED
+ */
+public class DebuggingExampleUrlClient {
+
+    static {
+        /*
+         * 1. (optional) Disables the AsyncContext associated with individual request/responses to reduce stack-trace
+         * depth and simplify execution tracing. This will disable/break some features such as request tracing,
+         * authentication, propagated timeouts, etc. that rely upon the AsyncContext so should only be disabled when
+         * necessary for debugging:
+         */
+        // AsyncContext.disable();
+    }
+
+    /**
+     * Log all protocol user data. Logging user data may expose sensitive content contained in the headers or payload
+     * body. Care and consideration should be taken before enabling this feature on production systems.
+     *
+     * <p>If {@link Boolean#TRUE} then all user data in the headers and payload bodies will be logged in addition to
+     * network events.
+     * <p>If {@link Boolean#FALSE} then only network events will be logged, but user data contents will be omitted.
+     *
+     * <p>This implementation uses a constant function to enable or disable logging of user data, your implementation
+     * could selectively choose at runtime to log user data based upon application state or context.</p>
+     */
+    static final BooleanSupplier LOG_USER_DATA = Boolean.TRUE::booleanValue;
+
+    public static void main(String... args) throws Exception {
+        final HostAndPort debuggingExampleServerAddress = HostAndPort.of("localhost", 8080);
+        try (HttpClient client = buildHttpClient(debuggingExampleServerAddress)) {
+            final String requestTarget = String.format("http://%s/sayHello", debuggingExampleServerAddress);
+            client.request(client.post(requestTarget).payloadBody("George", textSerializerUtf8()))
+                    .whenOnSuccess(resp -> {
+                        System.out.println(resp.toString((name, value) -> value));
+                        System.out.println(resp.payloadBody(textSerializerUtf8()));
+                    })
+            // This example is demonstrating asynchronous execution, but needs to prevent the main thread from exiting
+            // before the response has been processed. This isn't typical usage for an asynchronous API but is useful
+            // for demonstration purposes.
+                    .toFuture().get();
+        }
+    }
+
+    private static HttpClient buildHttpClient(final HostAndPort debuggingExampleServerAddress) {
```

Comment:
I would propose to remove the whole `buildHttpClient` method and inline all logic inside `main`, to keep examples consistent.